src/config.rs

```diff
@@ -57,13 +57,29 @@ impl FlokiConfig {
             error: e,
         })?;
 
-        let config = serde_yaml::from_reader(f).map_err(|e| {
+        let mut config: FlokiConfig = serde_yaml::from_reader(f).map_err(|e| {
             errors::FlokiError::ProblemParsingConfigYaml {
                 name: file.display().to_string(),
                 error: e,
             }
         })?;
 
+        // Ensure the path to an external yaml file is correct.
+        // If the image.yaml.path file is relative, then it should
+        // be relative to the floki config file. At this point we
+        // already have the path to the floki config file, so we
+        // just prepend that to image.yaml.path.
+        if let image::Image::Yaml { ref mut yaml } = config.image {
+            if yaml.file.is_relative() {
```

Comment:
Probably the path wants to always be relative, since it probably ought to be part of the same codebase. I guess there is nothing to stop it being absolute though (and maybe that allows some nice abuse).

Reply:
Yeah I figured no-one would *want* to use an absolute path, but it doesn't hurt to check.